src/ipc/ipc4/helper.c

```diff
@@ -453,17 +453,42 @@ int ipc_comp_connect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 	if (!cpu_is_me(source->ipc_config.core) && !cross_core_bind)
 		return ipc4_process_on_core(source->ipc_config.core, false);
 
-	/* these might call comp_ipc4_get_attribute_remote() if necessary */
-	ret = comp_get_attribute(source, COMP_ATTR_BASE_CONFIG, &source_src_cfg);
-	if (ret < 0) {
-		tr_err(&ipc_tr, "failed to get base config for module %#x", dev_comp_id(source));
-		return IPC4_FAILURE;
+	struct processing_module *srcmod = comp_get_drvdata(source);
+	struct processing_module *dstmod = comp_get_drvdata(sink);
+	struct module_config *dstcfg = &dstmod->priv.cfg;
+	struct module_config *srccfg = &srcmod->priv.cfg;
```

Comment:
Something missing in the patch, "struct processing_module" definition is not known and there are lot of fails in CI (defined in module/base.h). Now this is supposed to be private to module implementations, which is why I think @ranj063  opted to add a new property so it can be "set from outside".

Reply:
@andyross I do like the idea of parsing/saving the base_cfg_ext data in the module adapter instead of doing it in the individual modules. But I think you're going to have to add a new property to retrieve the extn data just like we do for the base_cfg data today.

Reply:
I see the failures, presumably that's just a header ordering issue?  Or if processing_module is an illegal header in some contexts, can you provide details?  It would be easy enough to wrap these behind an API.  That seems unlikely though, as the alternate code was already banging directly on the mod->priv.cfg.

@ranj063 you mean the COMP_ATTR_BASE_CONFIG_EXT from the version in your first PR?  That's actually deliberately omitted: the only spot that would use it is using the original/kernel-derived data in the new parsed fields.    The reason for making this a property would presumably be that components could then override it, but (and, OK, this is becoming a crusade of mine I guess) I think that's emphatically bad and should be avoided.  If topology lied and gave us an invalid value, the job of the firmware is to fail gracefully and get the integrator to fix the topology.

Reply:
@andyross I think originally processing_module was intended to be private data accessible only for the module itself. But given that we are already accessing the base_cfg and base_cfg_ext are being accessed by the common code, I guess it makes sense to make it shared. By the same argument maybe we should remove the COMP_ATTR_BASE_CONFIG attribute as well.

Reply:
Agreed, I'll look at that as a followup cleanup.  The "kpb" component is the only spot I see in current code that's implementing the BASE_CONFIG attribute, and it's just echoing back the default base config anyway.